helix-view/src/document.rs

```diff
@@ -143,6 +144,7 @@ pub struct Document {
     language_server: Option<Arc<helix_lsp::Client>>,
 
     diff_handle: Option<DiffHandle>,
+    version_control_head: Option<Arc<ArcSwap<Arc<str>>>>,
```

Comment:
It seems weird to me that this is an ArcSwap wrapped in an Arc. Shouldn't we simply have an `ArcSwap<Arc<str>>`?

I think ArcSwap<T> already implies ArcSwapAny<Arc<T>> so actually just `ArcSwap<str>` might be enough.

Reply:
Ah that is my bad I suggested that but I mixedup `ArcSwapAny` and `ArcSwap` which aliases to `ArcSwapAny<Arc<T>`

Reply:
Thank you for the feedback! This is a great observation! I changed the type to `Option<Arc<ArcSwap<Box<str>>>>`. Unfortunately ArcSwap doesn't accept `str` directly because the size of the `str` values are not known at compilation time. We thought about choosing between `String` and `Box<str>` and went for the second, because that will be smaller.

Reply:
it';s really unfortunate that it doesn't allow that, see https://docs.rs/arc-swap/latest/arc_swap/docs/limitations/index.html for more details. We could use a `ThinaArc` here but it's probably not worth the extra dependency